File: server/app/api/nlp.py
from flask import Blueprint, request, jsonify
from app.models.nlp_model import tokenize_text

nlp_blueprint = Blueprint('nlp', __name__, url_prefix='/nlp')

@nlp_blueprint.route('/tokenize', methods=['POST'])
def tokenize():
    content = request.json
    text = content.get('text')
    tokens = tokenize_text(text)
    return jsonify(tokens=tokens)

File: client/src/lib/nlp.js
// lib/nlp.js
function tokenizeText(text) {
  return fetch('http://localhost:5000/nlp/tokenize', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ text: text }),
  })
  .then(response => response.json())
  .then(data => data.tokens);
}
